addon-test-support/-private/navigator.ts

```diff
@@ -0,0 +1,33 @@
+import RSVP from 'rsvp';
+import { proxyService } from 'ember-browser-services/utils/proxy-service';
+import { testableVersionOf } from './-helpers';
+
+export type TestNavigator = typeof fakeNavigator;
+
+type UserInteractions<Source> = Partial<Record<keyof Source, RSVP.Deferred<unknown>>>;
+
+export const fakeMediaDevices = testableVersionOf(
+  navigator.mediaDevices,
+  class TestMediaDevices {
+    interactions: UserInteractions<MediaDevices> = {};
```

Comment:
This is something I wanted to throw out there as an idea, but I don't think this library should start implementing this pattern (of adding stub / test-specific implementations of APIs) because the API surface is too great. 

Maybe we could extract this a bit to be more generic, and not tied to any one particular API?

would def be for a future PR to explore:
```ts
import { interaction, interactionFor } from 'ember-browser-services/test-support';

// ...
module('....', function (hooks) {
  setupBrowserFakes(hooks, {
     window: {
       Notification: {
           // to mimic the real implementation of requestPermission, the result must set a permission property
          @interaction((notification, ...args) => notification.permission = args[0])
          requestPermission;
       }
     }
  });
  
  test('...', async function(assert) {
     let service = this.owner.lookup('service:browser/window');
   
     // some component does:
     service.Notification.requestPermission(result => {
        assert.equal(result, 'granted');
     });  
     
     // we can assert the pending state, which normally would be impossible, due to user interaction
     // or all the caveats of swapping browser APIs at runtime
     assert ...
     
     // progress the interaction
     interactionFor(service.Notification.requestPermission).resolve('granted');
     
     assert..., 'user allowed notifications')
   });
});
```